corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -0,0 +1,87 @@
+import os
+from tempfile import mkstemp
+import uuid
+from django.core.cache import caches
+from corehq.apps.case_importer.tracking.models import CaseUploadFileMeta
+from corehq.blobs import get_blob_db
+from corehq.util.files import file_extention_from_filename
+from dimagi.utils.decorators.memoized import memoized
+
+BUCKET = 'case_importer/upload_files'
+
+
+class PersistentFileStore(object):
+    def __init__(self, bucket, meta_model):
+        """
+        :meta_model is a django model used to store meta info
+            must contain columns identifier, filename, length
+        """
+
+        self._bucket = bucket
+        self._db = get_blob_db()
+        self._meta_model = meta_model
+
+    def write_file(self, f, filename):
+        blob_info = self._db.put(f, bucket=self._bucket)
```

Comment:
> I was thinking of (bucket, identifier) as being roughly equivalent to couchdb's (database, _id)

(bucket, identifier) is more like (database + _id, attachment_name), although not exactly.

Yes, that sounds like a fine solution. Maybe keep the bucket prefix on the shorter side if there will be many of these since the bucket prefix is used in the key, which adds storage (and RAM for some configurations) overhead for the blob db.